enhancements/machine-api/dedicated-instances.md

```diff
@@ -0,0 +1,129 @@
+
+- Add ability to enable dedicated instances using Machine's provider spec.
+
+- Validate that provider spec doesn't contain spot instances configuration and dedicated instances at the same time when it's not supported by the cloud provider.
+
+- Make sure that some required changes are supported and implemented on OCP installer side, like proper control plane creation.
+
+### Implementation Details
+
+For each of three cloud providers that support dedicated instances the implementation will be different.
+
+#### AWS
+
+`Dedicated Instances are Amazon EC2 instances that run in a virtual private cloud (VPC) on hardware that's dedicated to a single customer.`. [AWS Documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/dedicated-instance.html). Each launched instance has a tenancy attribute and it can be configured similar to how we set availability zone.
+
+```
+placement = &ec2.Placement{
+  AvailabilityZone: aws.String(machineProviderConfig.Placement.AvailabilityZone),
+  Tenancy: aws.String(machineProviderConfig.Placement.Tenancy)
+}
+```
+
+That change will require adding `Tenancy` field to provider spec.
+
+#### Azure
+
+In order to make dedicated VMs work on Azure we need to understand the concept of host groups and hosts.
+[Azure documentation](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/dedicated-hosts).
+
+```
+A host group is a resource that represents a collection of dedicated hosts. You create a host group in a region and an availability zone, and add hosts to it.
+
+A host is a resource, mapped to a physical server in an Azure data center. The physical server is allocated when the host is created. A host is created within a host group. A host has a SKU describing which VM sizes can be created. Each host can host multiple VMs, of different sizes, as long as they are from the same size series.
+
+When creating a VM in Azure, you can select which dedicated host to use for your VM. You have full control as to which VMs are placed on your hosts.
+```
+
+Above means that we are required to have a host group and host created, this should probably be handled on installer side. 
+
+The problem here are standard quotas: for host of type `DSv3-Type1` we can create only 32 VM of type `Standard_D2s_v3`(default type for worker VMs). To request a quota increase, the users are required to create a support request. This part should be well documented
+
+The required API change is adding host name field `Host` to provider spec.
+
+#### GCP
+
+GCP provides `Node Templates`, `Node Groups` for managing dedicated instances. Similar to Azure, these should be created using OCP installer. [GCP documentation](https://cloud.google.com/compute/docs/nodes/sole-tenant-nodes).
+
+In order to be able to create a VM on a dedicated we should introduce `NodeGroup` API field to provider spec.
+
+We should document that node groups have certain capacity and amount of VMs is limited, unless the node group autoscaler is enabled.
```

Comment:
this part is a little confusing to me, does this mean that the node groups have both limited number of VMs and a resource capacity(eg ram, cpu) as well?

Reply:
The number of VMs is limited by node group capacity. You can create as many VMs as you want until the resources limit is reached.

Reply:
perhaps this makes it a little clearer?

```suggestion
We should document that node groups have resource capacities which limit the number of VMs, unless the node group autoscaler is enabled.
```